apps/severewxalertsusa/severewxalertsusa.star

```diff
@@ -0,0 +1,346 @@
+WARNING_IMG = base64.decode("iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAABABJREFUeF7t3G1u1DAQBuBUojeBU8DxyyngJiBRpSKrNBvb8/HOeGwPv1CbeNN5/I6dbLsvW/4LVYGXUFeTF7MlSLBJEAnkX6faRKpB14T0Ami5dwXq8eJRIa5QPWrjlpBREErpccOxfqHRIdxTYwkyG8aBY1kzk5Y1K4RLWtDabhhvX77e9vsff3+3dlHI76PrB0uIKUSp+NTKOiDBYBADwTG0AC0oIyBELdUJgWJYQ1yhDGDUKJoBIBjeCKX0AHE0NRUnRI0RBcIoNWIUyYnTYhw4oLRIastOiAojaioM2xgbhXPCUhjAtHBqTE6IGGO0VBilhYxCOXB5DFBaKLUmJUQEMksygLswCEhi3PQwxS6siVI7IDEqz2CsUKAgs7Yp8EJfTUnpm+x0rIahXOiLKBCQVTEUKCwQVjpWx0Cj3EklSOsNFczO6zYl1y8mhgADmRIxSO9W9f3Pr2rpfr5+U5RWfipzO/yUkvMXhkrHJCC7/CcUEUjvdOw/RVSQ/do0KUkQeXcqnukKEiEd0ROiSQk7IQlCi5Q0JSyQKBgjJISZkodDgtAmvOgoRkr4IJHSMUpCGClJENGUF5zETQm5ZWVCBBq8e5IPCxJINIyRWha3bSWIbOKzziK2rUwIq6qKgxNk27ZeT3vv3BJkZpCIC/poizpxYaetIQmiWDxOpxLaVoJgSk0bJUE6vYVb4kmQBKFFF3VU5LdwhVtf2hqyDx5xYR8JhNCuHo+x8tEJKrKVcZYAcagj7CUSBFZKzEAJgqkjbBQ4SNSFHVYxw4GIGLxFPUHkYgkir53JmUQQ/i85ZEr4XkSMR7v69J9t25q//R7xBpFfJr8zzEEipST6nToDQ56QBKGnyw0kCkrkhEgx2GvIMT8irCUJcklrb5SoIMx0XEPx9GlAzZ1WlJRMAlL9o8+91mSQ3mtJRBBtOp7i8n/2D4ESDQSBMTQIfQPqc6QlyFCty6fc9VdBYZQSwgbpvZ70RBFg1Ope/cxF1lqyIgoaoyrF3XFF2Q57pUWI0ap581NJ2SlZISlWGE0tyTZ49qRYYpiCzJgUBQa11s2WdUx4UeuaCcUDg6ymaV2jtzAlBLfG5ISokzJiWrwx2HqIpIwAA4CQ1padEEhSIqP0xBAropISCQYEoa2pOCGwpJzvrL3fhQQiHD9G9XPdKU8R1ANIH7HULs4axgBCnQyYKLJ9lZC0QEYA58tFTOyP8WADecBQ2ptD8U0g0Ak5X6T4rp7SYwMdg57MJglZAcYEwjIhM6OYYlisIaWOMnobM4fwSsgd0Cg4bgimuwTGohsVpgtEz4REa2tdAa7FCHUxN1LoFEX/eeE3hoyOlYfeVSD8jFmNLUGCiSdIMJB3Az+ndAmyQLEAAAAASUVORK5CYII=")
+
+## run the main applications
+def main(config):
+    jsonLocation = json.decode(config.str("location") or DEFAULT_LOCATION)  ## set the location from the schema data or use the default
+
```

Comment:
Would you mind adding some code to truncate latitude and longitude? This would help protect against leaking users' exact location to the API provider. As a secondary benefit, it also improves cache rate a bit in densely populated areas.

I'm thinking 2 decimal points would be sufficient, based on [this discussion](https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude).

You can find some examples over here: https://github.com/tidbyt/community/search?q=truncate

Reply:
Great, thank you!

There is one small issue though. Shorter inputs do not appear to be truncated correctly. E.g., `truncated_location_digits("0.313") == "0.313"`, while `truncated_location_digits("0.3130") == "0.3150"`.

I think you can simplify the function and get the desired effect like this:
```
def truncated_location_digits(x):
    return str(int(math.round(float(x) * 200)) / 200)
```

Reply:
To unblock, I am going to merge and update main with the suggestions